tools/pkg/engine_build_configs/lib/src/build_config.dart

```diff
@@ -0,0 +1,820 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+// This library parses Engine build config data out of the "Engine v2" build
+// config JSON files with the format described at:
+//   https://github.com/flutter/engine/blob/main/ci/builders/README.md
+
+/// Base class for all nodes in the build config.
+base class BuildConfigBase {
```

Comment:
optional (maybe file an issue to do it later): We could make this `sealed base` to allow exhaustive checking/pattern matching of the different kinds of BuildConfigs! It wouldn't have any other changes to your code I believe.

Reply:
Changed to `sealed`.